src/EditorFeatures/Core/Implementation/IntelliSense/Completion/AsyncCompletionService.cs

```diff
@@ -87,10 +87,28 @@ private bool UseLegacyCompletion(ITextView textView, ITextBuffer subjectBuffer)
         {
             if (!_newCompletionAPIEnabled.HasValue)
             {
-                if (Workspace.TryGetWorkspace(subjectBuffer.AsTextContainer(), out var workspace))
+                int userSetting = 0;
+                const string useAsyncCompletionOptionName = "UseAsyncCompletion";
+                if (textView.Options.GlobalOptions.IsOptionDefined(useAsyncCompletionOptionName, localScopeOnly: false))
                 {
-                    var experimentationService = workspace.Services.GetService<IExperimentationService>();
-                    _newCompletionAPIEnabled = experimentationService.IsExperimentEnabled(WellKnownExperimentNames.CompletionAPI);
+                    userSetting = textView.Options.GlobalOptions.GetOptionValue<int>(useAsyncCompletionOptionName);
+                }
+
+                if (userSetting == 1)
+                {
+                    _newCompletionAPIEnabled = true;
+                }
+                else if (userSetting == -1)
+                {
+                    _newCompletionAPIEnabled = false;
+                }
```

Comment:
i don't really understand where teh 1 and -1 values came from...   How do you know it will be those values.  Do we have code elsewhere that sets these?  If so, can we define constants somewhere that we can share between the reading/writing code?

Reply:
Good catch! Thank you, @CyrusNajmabadi! Let me add a comment

Reply:
It corresponds to values of a three-state checkbox

Reply:
and it is an convention used in VS settings